src/tests/ftest/util/server_utils.py

```diff
@@ -50,6 +58,422 @@ class ServerFailed(Exception):
+        }
+        start_time = time.time()
+        start_msgs = 0
+        timed_out = False
+        while start_msgs != self.server_cnt and not timed_out:
+            output = sub_process.get_stdout()
+            start_msgs = len(re.findall(patterns[self.mode], output))
+            timed_out = time.time() - start_time > self.timeout
+
+        if start_msgs != self.server_cnt:
+            err_msg = "{} detected. Only {}/{} messages received".format(
+                "Time out" if timed_out else "Error",
+                start_msgs, self.server_cnt)
+            self.log.info("%s:\n%s" % (err_msg, sub_process.get_stdout()))
+            return False
+
+        self.log.info("Started server in <%s> mode in %d seconds" % (self.mode,
+            time.time() - start_time))
+        return True
+
+    class DaosServerConfig(ObjectWithParameters):
+        """Object to manage the configuration of the server command."""
+        # pylint: disable=pylint-no-self-use
+
+        def __init__(self):
+            """ Create a DaosServerConfig object."""
+            super(DaosServer.DaosServerConfig, self).__init__(
+                "/run/server_config/*")
+            self.data = None
+            self.name = BasicParameter(None, "daos_server")
+            self.port = BasicParameter(None, "10001")
+            self.nvme = None
+            self.scm = None
+
+        @property
+        def name(self):
+            """Get the name from the server config."""
+            if self.data and "name" in self.data:
+                return self.data["name"]
+            return None
+
+        @name.setter
+        def name(self, value):
+            """Set the server config name attribute."""
+            if self.data and "name" in self.data:
+                self.data["name"] = value
+
+        @property
+        def port(self):
+            """Get the port from the server config."""
+            if self.data and "port" in self.data:
+                return self.data["port"]
+            return None
+
+        @port.setter
+        def port(self, value):
+            """Set the port config attribute."""
+            if self.data and "port" in self.data:
+                self.data["port"] = value
+
+        def get_params(self, test):
+            """Get values for all of the command params from the yaml file.
+
+            If no key matches are found in the yaml file the BasicParameter
+            object will be set to its default value.
+
+            Args:
+                test (Test): avocado Test object
+            """
+            super(DaosServer.DaosServerConfig, self). get_params(test)
+            # Read the baseline conf file data/daos_server_baseline.yml
+            try:
+                with open('{}/{}'.format(test.basepath, DEFAULT_FILE), 'r') \
+                as rfile:
+                    self.data = yaml.safe_load(rfile)
+            except Exception as err:
+                self.log.info("<SERVER> Exception occurred: %s" % str(err))
```

Comment:
(pylint-logging-not-lazy) Specify string format arguments as logging function parameters